src/CarbonAware.CLI/src/Commands/Emissions/EmissionsCommand.cs

```diff
@@ -41,17 +70,60 @@ private async Task Run(InvocationContext context)
         var locations = context.ParseResult.GetValueForOption<string[]>(_requiredLocation);
         var startTime = context.ParseResult.GetValueForOption<DateTimeOffset?>(_startTime);
         var endTime = context.ParseResult.GetValueForOption<DateTimeOffset?>(_endTime);
-        
-        var parameters = new CarbonAwareParametersBaseDTO() { 
-            MultipleLocations = locations,
+        var best = context.ParseResult.GetValueForOption<bool>(_best);
+        var average = context.ParseResult.GetValueForOption<bool>(_average);
+
+        var parameters = new CarbonAwareParametersBaseDTO()
+        {
             Start = startTime,
             End = endTime
         };
-
         // Call the aggregator.
-        var results = await aggregator.GetEmissionsDataAsync(parameters);
+        if (best)
+        {
+            parameters.MultipleLocations = locations;
+
+            var result = await aggregator.GetBestEmissionsDataAsync(parameters);
+           
+            if(result != null)
+            {
+                var serializedOuput = JsonSerializer.Serialize((EmissionsDataDTO)result);
```

Comment:
🐑 why do we need to do the casting here for the serializer but not in the case below. Can we not just enforce that the type of result is EmissionsDataDTO?

Reply:
In the case below we directly create an EmissionsDataDTO object hence dont have to cast. We can't enforce that the result is of type EmissionsDataDTO since it is the aggregator call which returns an EmissionsData object